torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -287,90 +287,107 @@ std::vector<std::shared_ptr<NCCLComm>>& ProcessGroupNCCL::getNCCLComm(
   return devNCCLCommMap_[devicesKey];
 }
 
-// Helper function that checks the input and output tensors for validity
-void ProcessGroupNCCL::tensorCheckHelper(
-    const std::vector<at::Tensor>& input,
-    const std::vector<at::Tensor>& output,
-    int outputOverInput) {
-  if (input.size() * outputOverInput != output.size()) {
-    throw std::runtime_error(
-        "Input tensor sequence should have the same "
-        "number of tensors as the output tensor sequence");
-  }
+namespace {
 
-  if (input.size() == 0) {
-    throw std::runtime_error("The number of input tensors should not be zero");
+/*
+ * Check that all `tensors' have the same type and shape and are distributed
+ * across distinct GPUs.
+ */
+void check_gpu_tensors(const std::vector<at::Tensor>& tensors) {
+  if (tensors.size() == 0) {
+    throw std::runtime_error("Tensor list must be nonempty");
   }
-
-  if (input.size() > static_cast<size_t>(at::cuda::getNumGPUs())) {
+  if (tensors.size() > static_cast<size_t>(at::cuda::getNumGPUs())) {
     throw std::runtime_error(
-        "The number of input tensors is larger than "
-        "the number of available GPUs");
+      "Tensor list mustn't be larger than the number of available GPUs"
```

Comment:
`clang-format` will reformat this with bigger indent

Reply:
Does PyTorch use clang-format as a source-of-truth?  I'm freaked out about what it does to lambdas.

Reply:
This file is decidedly not clang-formatted, and running clang-format indeed turns the lambda formatting into spaghetti.  I'll leave it for now and defer to the code formatting zealots who surely will visit upon this file if they find it necessary at some point 😬

Reply:
There are a few other clang-format violations in c10d right now, so we can address this in a separate commit. Ideally I'd like to setup a clang-format linter for this directory (and others where applicable) so we don't have to deal with partially formatted files.